### DISCUSSION TITLE
Auto update r3f scene on edit

### DISCUSSION THREAD

**Discussion by patrik-u** [splinetool/r3f-spline#10] [Q&A]
Hi, I'm looking for a tool that will help with my r3f workflow and came across Spline as a potential solution. I'd like to be able to edit a scene in visual editor, save it and at most do a page refresh to see the updates. r3f-spline seems somewhat promising but it looks like I have to export the code, copy and paste it each time I update the scene, which is a bit clunky. It would be nice if I could just have the scene automatically generated by from the .splinecode file. Is this something that I can do today by iterating through the nodes? Or is this something that is in the works?

**Comment by marcofugaro** [comment]
You can use the [Clone](https://github.com/pmndrs/drei#clone) component from drei to avoid writing the whole scenegraph declaratively.

You can find an example usage in [this r3f demo](https://codesandbox.io/s/f79ucc)
